src/lti/LTI_Message_Launch.php

```diff
@@ -70,50 +70,18 @@ public static function from_cache($launch_id, Database $database, Cache $cache =
     /**
      * Validates all aspects of an incoming LTI message launch and caches the launch if successful.
      *
-     * @param array|string $request An array of post request parameters. If not set will default to $_POST.
-     *
      * @return LTI_Message_Launch   Will return $this if validation is successful.
-     * @throws LTI_Exception        Will throw an LTI_Exception if validation fails.
+     * @throws LTI_Exception Will throw an LTI_Exception if validation fails.
      */
-    public function validate(array $request = null)
+    public function validate(bool $skip_state = false, bool $allow_self_signed = false)
```

Comment:
ive given up on this code ever being merged back to the IMS repo, so i went ahead and removed the unused $request param on this method.